Sim/SimPapas/src/PapasAlg.h

```diff
@@ -35,16 +41,24 @@ class PapasAlg : public GaudiAlgorithm {
   virtual StatusCode finalize();
 
 private:
+  std::shared_ptr<papas::Detector> m_spDetector;  ///< papas detector
   /// Stores the linkages between papas objects (eg clusters, tracks, particles)
   papas::Nodes m_history;
-  std::vector<IPapasTool*> m_tools;      ///<vector of tools to be run
-  std::vector<std::string> m_toolNames;  ///< names of tools to be run
-  long m_eventno;                        ///< the papas Event number, incremented for each event processed
+  std::vector<IPapasTool*> m_tools;       ///< vector of tools to be run
+  IPapasImportTool* m_importTool;         ///< import tool to be run
+  IPapasExportTool* m_exportTool;         ///< export tool to be run
+  std::vector<std::string> m_toolNames;   ///< names of tools to be run
```

Comment:
why are the names necessary?
for m_tools, couldn't you use a 

```c++
std::vector< std::pair<std::string, IPapasTool*> >
```

instead?

Reply:
If I understand correctly, its not possible to pass any old argument through from the python configuration file (such as your vector above), but one can pass a list of strings. So the PapasAlg takes a list of tool names and then the individual tools are picked up inside the Algorithm. My approach is based on similar code in other parts of FCCSW

Reply:
ok